src/java/org/apache/cassandra/transport/ConnectionLimitHandler.java

```diff
@@ -61,7 +61,7 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception
         if (count > limit)
         {
             // The decrement will be done in channelClosed(...)
-            noSpamLogger.warn("Exceeded maximum native connection limit of {} by using {} connections", limit, count);
+            noSpamLogger.error("Exceeded maximum native connection limit of {} by using {} connections", limit, count);
```

Comment:
I'm not sure how to use the noSpamLogger globally, because it looks like loggings that have been confusing folks a lot. You see error messages there that you can ignore (like the one with allocation for the chunk cache being impossible, which is not something that's really actionable).
What's the intent behind the noSpamLogger and what are operators supposed to do about the messages you're raising from warn to error here?
Is that ConnectionLimitHandler used in incremental repair?

Reply:
The nospam logger has some logic to prevent flooding logs with too much information.  The comment on the class says this:

```
Logging that limits each log statement to firing based on time since the statement last fired.
```

You'd see this if you had misconfigured your client to make too many connections to the DB.  For example if you did `.setConnectionsPerHost(1000)`.

Reply:
ok gotcha. 

But if I run into such a log message, what should I do as an operator?
Maybe we could give some pointers on how to remediate the situation? Do you have some safe recommendations to make here (like the driver config you mentioned).

Reply:
OK.  I've added some extra information to the logs which should make the origin a little more clear.  

One thing to note is that this message can only occur if someone has explicitly set native_transport_max_concurrent_connections, or native_transport_max_concurrent_connections_per_ip.